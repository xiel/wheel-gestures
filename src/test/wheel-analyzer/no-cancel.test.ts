import macSafariLeft from '../fixtures/no-cancel/mac-swipe-left-end-bounce.json'
import winRightFast from '../fixtures/no-cancel/win-chrome-fast-right-ptp.json'
import winRight from '../fixtures/no-cancel/win-chrome-right-ptp.json'
import { recordPhases } from '../helper/recordPhases'

describe('no cancel', () => {
  describe('slight increase in delta should not trigger momentum cancel', () => {
    test('Mac - Safari', () => {
      expect(recordPhases(macSafariLeft.wheelEvents)).toMatchSnapshot()
    })

    test('Win - Chrome - swipe right', () => {
      expect(recordPhases(winRight.wheelEvents)).toMatchSnapshot()
    })

    test('Win - Chrome - swipe right fast', () => {
      expect(recordPhases(winRightFast.wheelEvents)).toMatchSnapshot()
    })
  })
})
