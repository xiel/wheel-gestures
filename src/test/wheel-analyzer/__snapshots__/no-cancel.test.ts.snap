// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no cancel slight increase in delta should not trigger momentum cancel Mac - Safari 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          2558,
          0,
          0,
        ],
        "axisVelocity": Array [
          3.606060606060606,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": 57,
          "deltaY": 0,
          "timeStamp": 47900,
        },
        "isMomentum": false,
        "type": "WHEEL_END",
      },
      "range": Array [
        0,
        23,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          3148,
          0,
          0,
        ],
        "axisVelocity": Array [
          0.04,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": 1,
          "deltaY": 0,
          "timeStamp": 48634,
        },
        "isMomentum": true,
        "type": "MOMENTUM_WHEEL_END",
      },
      "range": Array [
        23,
        65,
      ],
      "wheelType": "MOMENTUM_WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          3148,
          0,
          0,
        ],
        "axisVelocity": Array [
          0.04,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": 1,
          "deltaY": 0,
          "timeStamp": 48634,
        },
        "isMomentum": true,
        "type": "ANY_WHEEL_END",
      },
      "range": Array [
        0,
        65,
      ],
      "wheelType": "ANY_WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -9222,
          0,
          0,
        ],
        "axisVelocity": Array [
          -26.393939393939394,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -425,
          "deltaY": 0,
          "timeStamp": 49110,
        },
        "isMomentum": false,
        "type": "WHEEL_END",
      },
      "range": Array [
        66,
        85,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -14998,
          0,
          0,
        ],
        "axisVelocity": Array [
          -0.04081632653061224,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -1,
          "deltaY": 0,
          "timeStamp": 50277,
        },
        "isMomentum": true,
        "type": "MOMENTUM_WHEEL_END",
      },
      "range": Array [
        85,
        152,
      ],
      "wheelType": "MOMENTUM_WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -14998,
          0,
          0,
        ],
        "axisVelocity": Array [
          -0.04081632653061224,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -1,
          "deltaY": 0,
          "timeStamp": 50277,
        },
        "isMomentum": true,
        "type": "ANY_WHEEL_END",
      },
      "range": Array [
        66,
        152,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`no cancel slight increase in delta should not trigger momentum cancel Win - Chrome - swipe right 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -1261.5,
          0,
          0,
        ],
        "axisVelocity": Array [
          -1.6656662663927713,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -27,
          "deltaY": 0,
          "timeStamp": 4111.99000000488,
        },
        "isMomentum": false,
        "type": "WHEEL_END",
      },
      "range": Array [
        0,
        19,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -1810.5,
          0,
          0,
        ],
        "axisVelocity": Array [
          -0.020010672361040757,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -1.5,
          "deltaY": 0,
          "timeStamp": 5711.669999989681,
        },
        "isMomentum": true,
        "type": "MOMENTUM_WHEEL_END",
      },
      "range": Array [
        19,
        95,
      ],
      "wheelType": "MOMENTUM_WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -1810.5,
          0,
          0,
        ],
        "axisVelocity": Array [
          -0.020010672361040757,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -1.5,
          "deltaY": 0,
          "timeStamp": 5711.669999989681,
        },
        "isMomentum": true,
        "type": "ANY_WHEEL_END",
      },
      "range": Array [
        0,
        95,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;

exports[`no cancel slight increase in delta should not trigger momentum cancel Win - Chrome - swipe right fast 1`] = `
Object {
  "phases": Array [
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -2307,
          0,
          0,
        ],
        "axisVelocity": Array [
          -3.7600716194702604,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -61.5,
          "deltaY": 0,
          "timeStamp": 38931.870000000345,
        },
        "isMomentum": false,
        "type": "WHEEL_END",
      },
      "range": Array [
        0,
        19,
      ],
      "wheelType": "WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -3631.5,
          0,
          0,
        ],
        "axisVelocity": Array [
          -0.025725127018650807,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -1.5,
          "deltaY": 0,
          "timeStamp": 40598.1900000188,
        },
        "isMomentum": true,
        "type": "MOMENTUM_WHEEL_END",
      },
      "range": Array [
        19,
        107,
      ],
      "wheelType": "MOMENTUM_WHEEL",
    },
    Object {
      "lastData": Object {
        "axisDelta": Array [
          0,
          0,
          0,
        ],
        "axisMovement": Array [
          -3631.5,
          0,
          0,
        ],
        "axisVelocity": Array [
          -0.025725127018650807,
          0,
          0,
        ],
        "event": Object {
          "deltaMode": 0,
          "deltaX": -1.5,
          "deltaY": 0,
          "timeStamp": 40598.1900000188,
        },
        "isMomentum": true,
        "type": "ANY_WHEEL_END",
      },
      "range": Array [
        0,
        107,
      ],
      "wheelType": "ANY_WHEEL",
    },
  ],
}
`;
